src/App.js

```diff
@@ -14,11 +16,105 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.PureComponent {
+  state = {
+    goodsList: [...goodsFromServer],
+    isStarted: false,
+  }
+
+  start = () => {
+    this.setState({ isStarted: true });
+  }
+
+  reverse = () => {
+    this.setState(
+      state => ({
+        goodsList: [...state.goodsList.reverse()],
```

Comment:
```suggestion
        goodsList: [...state.goodsList].reverse(),
```
тебе сначала нужно сделать копию, а потом уже ее менять.